src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,33 @@
+import { useParams } from "react-router-dom";
+import { useLocation } from "react-router-dom";
+
+function SingleArticle() {
+  const { article_id } = useParams();
+  const location = useLocation();
+  const articles = location.state;
+
+ console.log(article_id)
+
+  for (let i = 0; i < articles.length; i++) {
+    if (article_id == articles[i].article_id) {
+      const article = articles[i];
```

Comment:
This is unnecessary logic. Rather than using the getAllArticles data, why not make a fresh API request for a single article. it requires less logic and doesn't tie you to the articles state. You have the endpoint so make use of it.